### ISSUE TITLE
Cannot put UICanvas from toolbox on Scene or create it in a script

### ISSUE THREAD

**Issue by Crawcik** [FlaxEngine/FlaxEngine#20]
**Issue description:**
- Cannot put UI Canvas in scene (dragged from Toolbox)
- Instantiate this in script

**Steps to reproduce:**
Just try drag and drop it on scene

Solution can be adding constructor to UICanvas (maybe)

**Error**
```Exception has been thrown during Window.OnDragDrop. Default constructor not found for type FlaxEngine.UICanvas
Stack strace:
  at System.RuntimeType.CreateInstanceMono (System.Boolean nonPublic, System.Boolean wrapExceptions) [0x0007b] in <ccba8f4931be433d979e22082f9d6862>:0 
  at System.RuntimeType.CreateInstanceSlow (System.Boolean publicOnly, System.Boolean wrapExceptions, System.Boolean skipCheckThis, System.Boolean fillCache, System.Threading.StackCrawlMark& stackMark) [0x00009] in <ccba8f4931be433d979e22082f9d6862>:0 
  at System.RuntimeType.CreateInstanceDefaultCtor (System.Boolean publicOnly, System.Boolean skipCheckThis, System.Boolean fillCache, System.Boolean wrapExceptions, System.Threading.StackCrawlMark& stackMark) [0x00027] in <ccba8f4931be433d979e22082f9d6862>:0 
  at System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic, System.Boolean wrapExceptions) [0x0002c] in <ccba8f4931be433d979e22082f9d6862>:0 
  at System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic) [0x00000] in <ccba8f4931be433d979e22082f9d6862>:0 
  at System.Activator.CreateInstance (System.Type type) [0x00000] in <ccba8f4931be433d979e22082f9d6862>:0 
  at FlaxEditor.Scripting.ScriptType.CreateInstance () [0x0000e] in C:\Flax\FlaxEngine\Source\Editor\Scripting\ScriptType.cs:818 
  at FlaxEditor.SceneGraph.GUI.ActorTreeNode.OnDragDropHeader (FlaxEngine.GUI.DragData data) [0x00665] in C:\Flax\FlaxEngine\Source\Editor\SceneGraph\GUI\ActorTreeNode.cs:680 
  at FlaxEditor.GUI.Tree.TreeNode.OnDragDrop (FlaxEngine.Vector2& location, FlaxEngine.GUI.DragData data) [0x0001c] in C:\Flax\FlaxEngine\Source\Editor\GUI\Tree\TreeNode.cs:979 
  at FlaxEngine.GUI.ContainerControl.OnDragDrop (FlaxEngine.Vector2& location, FlaxEngine.GUI.DragData data) [0x00047] in C:\Flax\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:1134 
  at FlaxEditor.GUI.Tree.TreeNode.OnDragDrop (FlaxEngine.Vector2& location, FlaxEngine.GUI.DragData data) [0x00000] in C:\Flax\FlaxEngine\Source\Editor\GUI\Tree\TreeNode.cs:969 
  at FlaxEngine.GUI.ContainerControl.OnDragDrop (FlaxEngine.Vector2& location, FlaxEngine.GUI.DragData data) [0x00047] in C:\Flax\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:1134 
  at FlaxEngine.GUI.ContainerControl.OnDragDrop (FlaxEngine.Vector2& location, FlaxEngine.GUI.DragData data) [0x00047] in C:\Flax\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:1134 
  at FlaxEngine.GUI.ContainerControl.OnDragDrop (FlaxEngine.Vector2& location, FlaxEngine.GUI.DragData data) [0x00047] in C:\Flax\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:1134 
  at FlaxEngine.GUI.ContainerControl.OnDragDrop (FlaxEngine.Vector2& location, FlaxEngine.GUI.DragData data) [0x00047] in C:\Flax\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:1134 
  at FlaxEngine.GUI.ContainerControl.OnDragDrop (FlaxEngine.Vector2& location, FlaxEngine.GUI.DragData data) [0x00047] in C:\Flax\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:1134 
  at FlaxEngine.GUI.ContainerControl.OnDragDrop (FlaxEngine.Vector2& location, FlaxEngine.GUI.DragData data) [0x00047] in C:\Flax\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:1134 
  at FlaxEngine.GUI.ContainerControl.OnDragDrop (FlaxEngine.Vector2& location, FlaxEngine.GUI.DragData data) [0x00047] in C:\Flax\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:1134 
  at FlaxEngine.GUI.ContainerControl.OnDragDrop (FlaxEngine.Vector2& location, FlaxEngine.GUI.DragData data) [0x00047] in C:\Flax\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:1134 
  at FlaxEngine.GUI.ContainerControl.OnDragDrop (FlaxEngine.Vector2& location, FlaxEngine.GUI.DragData data) [0x00047] in C:\Flax\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:1134 
  at FlaxEngine.GUI.ContainerControl.OnDragDrop (FlaxEngine.Vector2& location, FlaxEngine.GUI.DragData data) [0x00047] in C:\Flax\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:1134 
  at FlaxEngine.GUI.ContainerControl.OnDragDrop (FlaxEngine.Vector2& location, FlaxEngine.GUI.DragData data) [0x00047] in C:\Flax\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:1134 
  at FlaxEngine.GUI.ContainerControl.OnDragDrop (FlaxEngine.Vector2& location, FlaxEngine.GUI.DragData data) [0x00047] in C:\Flax\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:1134 
  at FlaxEngine.Window.Internal_OnDragDrop (FlaxEngine.Vector2& mousePos, System.Boolean isText, System.String[] data) [0x00027] in C:\Flax\FlaxEngine\Source\Engine\Platform\Window.cs:382
```

**Flax version:**
1.0

**Comment by doshaq** [comment]
I solved this by right clicking the root and adding it that way, rather than pulling it.
![image](https://user-images.githubusercontent.com/15630572/102717741-4101f480-42f5-11eb-907b-d4dbd445d2ec.png)

**Comment by Crawcik** [comment]
> 
> 
> I solved this by right clicking the root and adding it that way, rather than pulling it.

Yeah, but still I can't spawn UICanvas in a script

**Comment by mafiesto4** [comment]
Fixed by 8680aba8971e06248a60e937d0013e312178b2e2
